core/metamodel.class.php

```diff
@@ -7095,32 +7095,41 @@ public static function BulkUpdate(DBObjectSearch $oFilter, array $aValues)
 	 */
 	public static function PurgeData($oFilter)
 	{
+		$iMaxBufferSize = MetaModel::GetConfig()->GetMaxBufferSize();
 		$sTargetClass = $oFilter->GetClass();
-		$oSet = new DBObjectSet($oFilter);
-		$oSet->OptimizeColumnLoad(array($sTargetClass => array('finalclass')));
-		$aIdToClass = $oSet->GetColumnAsArray('finalclass', true);
-
-		$aIds = array_keys($aIdToClass);
-		if (count($aIds) > 0)
-		{
-			$aQuotedIds = CMDBSource::Quote($aIds);
-			$sIdList = implode(',', $aQuotedIds);
-			$aTargetClasses = array_merge(
-				self::EnumChildClasses($sTargetClass, ENUM_CHILD_CLASSES_ALL),
-				self::EnumParentClasses($sTargetClass, ENUM_PARENT_CLASSES_EXCLUDELEAF)
-			);
-			foreach($aTargetClasses as $sSomeClass)
-			{
-				$sTable = MetaModel::DBGetTable($sSomeClass);
-				$sPKField = MetaModel::DBGetKey($sSomeClass);
+		$iNbIdsDeleted = 0;
+		$bExecuteQuery = true;
+
+		while ($bExecuteQuery) {
+			$oSet = new DBObjectSet($oFilter, array(), array(), null, $iMaxBufferSize);
```

Comment:
Rather than passing many default values, you can use the `SetLimit()`:

```suggestion
			$oSet = new DBObjectSet($oFilter);
			$oSet->SetLimit($iMaxBufferSize);
```